applications/API/serializers.py

```
@@ -0,0 +1,565 @@
+                "category_name": value,
+            }
+        except serializers.ValidationError:
+            try:
+                return {
+                    "category_key": value,
+                    "category_name": type(self).cacher.get_category_name_by_key(value),
+                }
+            except serializers.ValidationError as e:
+                raise serializers.ValidationError(
+                    f"Category '{value}' not found. Please provide a valid category key or name."
+                ) from e
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)